src/hevm/src/EVM/VMTest.hs

```diff
@@ -114,72 +116,77 @@ checkExpectation diff x vm = do
   return okState
 
 -- quotient account state by nullness
-(~=) :: Map Addr EVM.Contract -> Map Addr EVM.Contract -> Bool
-(~=) cs cs' =
+(~=) :: Map Addr (EVM.Contract, Storage) -> Map Addr (EVM.Contract, Storage) -> Bool
+(~=) cs1 cs2 =
     let nullAccount = EVM.initialContract (EVM.RuntimeCode mempty)
-        padNewAccounts cs'' ks = (fold [Map.insertWith (\_ x -> x) k nullAccount | k <- ks]) cs''
-        padded_cs' = padNewAccounts cs' (Map.keys cs)
-        padded_cs  = padNewAccounts cs  (Map.keys cs')
-    in and $ zipWith (===) (Map.elems padded_cs) (Map.elems padded_cs')
-
-(===) :: EVM.Contract -> EVM.Contract -> Bool
-a === b = codeEqual && storageEqual && (view balance a == view balance b) && (view nonce a == view nonce b)
+        padNewAccounts cs ks = Map.union cs $ Map.fromList [(k, (nullAccount, mempty)) | k <- ks]
+        padded_cs1 = padNewAccounts cs1 (Map.keys cs2)
+        padded_cs2 = padNewAccounts cs2 (Map.keys cs1)
+    in and $ zipWith (===) (Map.elems padded_cs1) (Map.elems padded_cs2)
+
+(===) :: (EVM.Contract, Storage) -> (EVM.Contract, Storage) -> Bool
+(c1, s1) === (c2, s2) =
+  codeEqual && storageEqual && (c1 ^. balance == c2 ^. balance) && (c1 ^. nonce ==  c2 ^. nonce)
   where
-    --storageEqual = view storage a == view storage b
-    storageEqual = undefined
-    codeEqual = case (view contractcode a, view contractcode b) of
+    storageEqual = s1 == s2
+    codeEqual = case (c1 ^. contractcode, c2 ^. contractcode) of
       (EVM.RuntimeCode a', EVM.RuntimeCode b') -> a' == b'
       _ -> error "unexpected code"
 
-checkExpectedContracts :: HasCallStack => EVM.VM -> Map Addr EVM.Contract -> (Bool, Bool, Bool, Bool, Bool)
+checkExpectedContracts :: HasCallStack => EVM.VM -> Map Addr (EVM.Contract, Storage) -> (Bool, Bool, Bool, Bool, Bool)
 checkExpectedContracts vm expected =
-  let cs = vm ^. EVM.env . EVM.contracts . to (fmap (clearZeroStorage.clearOrigStorage))
-      expectedCs = clearOrigStorage <$> expected
+  let cs = zipWithStorages $ vm ^. EVM.env . EVM.contracts -- . to (fmap (clearZeroStorage.clearOrigStorage))
+      expectedCs = clearStorage <$> expected
   in ( (expectedCs ~= cs)
      , (clearBalance <$> expectedCs) ~= (clearBalance <$> cs)
      , (clearNonce   <$> expectedCs) ~= (clearNonce   <$> cs)
      , (clearStorage <$> expectedCs) ~= (clearStorage <$> cs)
      , (clearCode    <$> expectedCs) ~= (clearCode    <$> cs)
      )
-
+  where
+  zipWithStorages = Map.mapWithKey (\addr c -> (c, lookupStorage addr))
+  lookupStorage addr =
+    case vm ^. EVM.env . EVM.storage of
+      ConcreteStore s -> mempty -- clearZeroStorage $ fromMaybe mempty $ Map.lookup (num addr) s
+      EmptyStore -> mempty
+      AbstractStore -> mempty -- error "AbstractStore, should this be handled?"
+      SStore {} -> mempty -- error "SStore, should this be handled?"
```

Comment:
what needs to be done here? I think we can safely error out on any symbolic storage expressions here since VMTests should always be concrete...